diff -uarN kdepim-15.08.0.orig/kleopatra/libkleopatraclient/CMakeLists.txt kdepim-15.08.0/kleopatra/libkleopatraclient/CMakeLists.txt
--- kdepim-15.08.0.orig/kleopatra/libkleopatraclient/CMakeLists.txt	2015-08-10 21:56:00.000000000 +0200
+++ kdepim-15.08.0/kleopatra/libkleopatraclient/CMakeLists.txt	2015-08-24 23:05:07.650076668 +0200
@@ -4,6 +4,11 @@
 add_subdirectory(core)
 add_subdirectory(gui)
 
+install(
+  FILES kleopatraclient_export.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/libkleopatraclient
+  )
+
 if(BUILD_TESTING)
     add_subdirectory(tests)
 endif()
diff -uarN kdepim-15.08.0.orig/kleopatra/libkleopatraclient/core/CMakeLists.txt kdepim-15.08.0/kleopatra/libkleopatraclient/core/CMakeLists.txt
--- kdepim-15.08.0.orig/kleopatra/libkleopatraclient/core/CMakeLists.txt	2015-08-10 21:56:00.000000000 +0200
+++ kdepim-15.08.0/kleopatra/libkleopatraclient/core/CMakeLists.txt	2015-08-24 22:52:15.613045160 +0200
@@ -67,3 +67,13 @@
 target_link_libraries(kleopatraclientcore Qt5::Widgets KF5::I18n)
 
 install(TARGETS kleopatraclientcore ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
+
+install(
+  FILES
+    initialization.h
+    command.h
+    selectcertificatecommand.h
+    signencryptfilescommand.h
+    decryptverifyfilescommand.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/libkleopatraclient/core
+  )
diff -uarN kdepim-15.08.0.orig/kleopatra/libkleopatraclient/gui/CMakeLists.txt kdepim-15.08.0/kleopatra/libkleopatraclient/gui/CMakeLists.txt
--- kdepim-15.08.0.orig/kleopatra/libkleopatraclient/gui/CMakeLists.txt	2015-08-10 21:56:00.000000000 +0200
+++ kdepim-15.08.0/kleopatra/libkleopatraclient/gui/CMakeLists.txt	2015-08-24 22:52:15.613045160 +0200
@@ -16,3 +16,9 @@
 )
 
 install(TARGETS kleopatraclientgui ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
+
+install(
+  FILES
+    certificaterequester.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/libkleopatraclient/gui
+  )
diff -uarN kdepim-15.08.0.orig/libkleo/CMakeLists.txt kdepim-15.08.0/libkleo/CMakeLists.txt
--- kdepim-15.08.0.orig/libkleo/CMakeLists.txt	2015-08-10 21:56:00.000000000 +0200
+++ kdepim-15.08.0/libkleo/CMakeLists.txt	2015-08-24 22:52:15.613045160 +0200
@@ -126,3 +126,61 @@
 else ()
 install ( FILES libkleopatrarc.desktop DESTINATION ${KDE_INSTALL_CONFDIR} RENAME libkleopatrarc )
 endif ()
+
+install( FILES
+    kleo/kleo_export.h
+    kleo/enum.h
+    kleo/oidmap.h
+    kleo/cryptobackend.h
+    kleo/cryptobackendfactory.h
+    kleo/cryptoconfig.h
+    kleo/dn.h
+    kleo/job.h
+    kleo/keylistjob.h
+    kleo/keygenerationjob.h
+    kleo/abstractimportjob.h
+    kleo/importjob.h
+    kleo/importfromkeyserverjob.h
+    kleo/exportjob.h
+    kleo/changeexpiryjob.h
+    kleo/changeownertrustjob.h
+    kleo/downloadjob.h
+    kleo/deletejob.h
+    kleo/encryptjob.h
+    kleo/decryptjob.h
+    kleo/signjob.h
+    kleo/specialjob.h
+    kleo/verifydetachedjob.h
+    kleo/verifyopaquejob.h
+    kleo/decryptverifyjob.h
+    kleo/signencryptjob.h
+    kleo/signkeyjob.h
+    kleo/adduseridjob.h
+    kleo/refreshkeysjob.h
+    kleo/multideletejob.h
+    kleo/hierarchicalkeylistjob.h
+    kleo/keyfilter.h
+    kleo/keyfiltermanager.h
+    kleo/changepasswdjob.h
+    kleo/checksumdefinition.h
+    kleo/exception.h
+    kleo/kconfigbasedkeyfilter.h
+    kleo/listallkeysjob.h
+    kleo/stl_util.h
+    DESTINATION ${INCLUDE_INSTALL_DIR}/kleo COMPONENT Devel)
+
+install ( FILES
+    ui/kdhorizontalline.h
+    ui/messagebox.h
+    ui/progressbar.h
+    ui/progressdialog.h
+    ui/keylistview.h
+    ui/keyselectiondialog.h
+    ui/keyrequester.h
+    ui/keyapprovaldialog.h
+    ui/dnattributeorderconfigwidget.h
+    ui/cryptoconfigmodule.h
+    ui/cryptoconfigdialog.h
+    ui/directoryserviceswidget.h
+    ui/filenamerequester.h
+    DESTINATION ${INCLUDE_INSTALL_DIR}/kleo/ui COMPONENT Devel)
diff -uarN kdepim-15.08.0.orig/libkpgp/CMakeLists.txt kdepim-15.08.0/libkpgp/CMakeLists.txt
--- kdepim-15.08.0.orig/libkpgp/CMakeLists.txt	2015-08-10 21:56:00.000000000 +0200
+++ kdepim-15.08.0/libkpgp/CMakeLists.txt	2015-08-24 23:04:02.505074051 +0200
@@ -44,6 +44,15 @@
 set_target_properties(kpgp PROPERTIES VERSION ${KDEPIM_LIB_VERSION} SOVERSION ${KDEPIM_LIB_SOVERSION})
 install(TARGETS kpgp ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
+install(FILES
+        kpgp.h
+        kpgpbase.h
+        kpgpblock.h
+        kpgpkey.h
+        kpgpui.h
+        libkpgp_export.h
+        DESTINATION ${INCLUDE_INSTALL_DIR}/kpgp COMPONENT Devel )
+
 if(BUILD_TESTING)
     add_subdirectory( tests )
 endif()
