commit 0dbc5c1b233d355dce26868253c0e8f4dede58d4
Author: Montel Laurent <montel@kde.org>
Date:   Mon Sep 5 16:01:10 2011 +0200

    Fix mem leak (too bad it will not in 4.7.1)

diff --git a/mailcommon/foldercollection.cpp b/mailcommon/foldercollection.cpp
index 3c31d8e..2d4ca73 100644
--- a/mailcommon/foldercollection.cpp
+++ b/mailcommon/foldercollection.cpp
@@ -294,8 +294,10 @@ uint FolderCollection::identity() const
     OrgKdeAkonadiImapSettingsInterface *imapSettingsInterface = Util::createImapSettingsInterface( mCollection.resource() );
     if ( imapSettingsInterface->isValid() ) {
       QDBusReply<bool> useDefault = imapSettingsInterface->useDefaultIdentity();
-      if( useDefault.isValid() && useDefault.value() )
+      if( useDefault.isValid() && useDefault.value() ) {
+	delete imapSettingsInterface;
         return mIdentity;
+      }
 
        QDBusReply<int> remoteAccountIdent = imapSettingsInterface->accountIdentity();
       if ( remoteAccountIdent.isValid() && remoteAccountIdent.value() > 0 ) {
